nelox/interpreter.py

```diff
@@ -258,6 +253,49 @@ def fn(*call_args):
                     env.set(var_name, value)
                     return value
 
+                elif name == "get-seq":
```

Comment:
There is a distinction worth making between two kinds of builtins:
1. Builtin functions work the same as other functions: you evaluate the arguments, then apply the function to the values.
2. Builtin special constructs do something with the expressions themselves, rather than with the values of those expressions.

For example, `(lambda (x y) body)` has to be a special construct, because you don't want to treat `(x y)` as an expression with a value.  However, you should generally prefer things to be functions, rather than special constructs, since they behave way more uniformly.

With this in mind, please take a look about which of these things need to be special constructs, while which can just be functions (and defined with your operators and such).